onebusaway-android/src/main/java/org/onebusaway/android/ui/TripDetailsListFragment.java

```diff
@@ -0,0 +1,352 @@
+            // Nothing to do right here...
+        }
+    }
+
+    private final static class TripDetailsLoader extends AsyncTaskLoader<ObaTripDetailsResponse> {
+
+        private final String mTripId;
+
+        TripDetailsLoader(Context context, String tripId) {
+            super(context);
+            mTripId = tripId;
+        }
+
+        @Override
+        public void onStartLoading() {
+            forceLoad();
+        }
+
+        @Override
+        public ObaTripDetailsResponse loadInBackground() {
+            return ObaTripDetailsRequest.newRequest(getContext(), mTripId).call();
+        }
+    }
+
+    private final class TripDetailsAdapter extends BaseAdapter {
+
+        LayoutInflater mInflater;
+
+        ObaTripSchedule mSchedule;
+        ObaReferences mRefs;
+        ObaTripStatus mStatus;
+
+        int mNextStopIndex;
+
+        public TripDetailsAdapter() {
+            this.mInflater = LayoutInflater.from(TripDetailsListFragment.this.getActivity());
+            this.mSchedule = mTripInfo.getSchedule();
+            this.mRefs = mTripInfo.getRefs();
+            this.mStatus = mTripInfo.getStatus();
+
+            String stopId = mStatus.getNextStop();
+            int i;
+            for (i = 0; i < mSchedule.getStopTimes().length; i++) {
+                ObaTripSchedule.StopTime time = mSchedule.getStopTimes()[i];
+                if (time.getStopId().equals(stopId)) {
+                    break;
+                }
+            }
+            mNextStopIndex = i;
+        }
+
+        @Override
+        public int getCount() {
+            return mSchedule.getStopTimes().length;
+        }
+
+        @Override
+        public Object getItem(int position) {
+            return mSchedule.getStopTimes()[position];
+        }
+
+        @Override
+        public long getItemId(int position) {
+            return position;
+        }
+
+        @Override
+        public View getView(int position, View convertView, ViewGroup parent) {
+            if (convertView == null) {
+                convertView = mInflater.inflate(R.layout.trip_details_listitem, parent, false);
+            }
+
+            ObaTripSchedule.StopTime stopTime = mSchedule.getStopTimes()[position];
+            ObaStop stop = mRefs.getStop(stopTime.getStopId());
+
+            TextView name = (TextView) convertView.findViewById(R.id.stop_name);
+            name.setText(stop.getName());
+
+            TextView time = (TextView) convertView.findViewById(R.id.time);
+            time.setText(DateUtils.formatDateTime(getActivity(),
+                    mStatus.getServiceDate() + stopTime.getArrivalTime() * 1000
+                            + mStatus.getScheduleDeviation() * 1000,
+                    DateUtils.FORMAT_SHOW_TIME |
+                            DateUtils.FORMAT_NO_NOON |
+                            DateUtils.FORMAT_NO_MIDNIGHT
+            ));
+
+            if (position < mNextStopIndex) {
```

Comment:
I like changing the styling of the item based on whether or not the bus has visited - that will be super useful.  However, I don't think the current styling difference of making the time bold/normal font is very obvious - its very easy to miss if you're not looking for it.

I'd prefer to see either the text color changed, or the background color changed for the stops that have already been visited.

I'd first try the text color change - any easy way to do this is to make the "passed" color transparent - for example, in the sliding panel header these two colors are used - one full white, and the other slightly transparent, to reduce emphasis on some of the lesser TextViews:

~~~
<color name="header_text_color">@android:color/white</color>
<color name="header_text_faded_color">#bbffffff</color>
~~~

On a related note, we should refresh this info occasionally - users will expect this information to update over time.  This should be fairly each to implement - you can following the existing model used in `ArrivalsListFragment`, and implement the same in `TripDetailsListFragment`.

You'll need the refresh handler:

~~~
private final Handler mRefreshHandler = new Handler();

    private final Runnable mRefresh = new Runnable() {
        public void run() {
            refresh();
        }
    };
~~~

...and a `refresh()` method:

~~~
private void refresh() {
        if (isAdded()) {
            UIHelp.showProgress(this, true);
            getArrivalsLoader().onContentChanged();
        }
    }
~~~

You'll also need to start the refresh in `onResume()`, and stop it in `onPause`.  Again, see `ArrivalsListFragment` for details.

And, we should also have a refresh icon in the Action Bar - see `menu/arrivals_list.xml` and `ArrivalsListFragment.onCreateOptionsMenu()` / `onOptionsItemSelected()` for how to implement.  

In the future (not for this PR), it would be good to add some kind of vehicle icon to show where the vehicle is add for each stop - we can tackle that later, though, when we do the other styling with the circles/lines for the route.

If you have any questions about any of this please let me know.